gateway-service/src/main/java/org/zowe/apiml/gateway/security/service/zosmf/AuthenticatedEndpointStrategy.java

```diff
@@ -75,7 +78,7 @@ private boolean endpointExists(TokenValidationRequest request, String endpoint)
         } else {
             return request.getEndpointExistenceMap().entrySet().stream()
                 .filter(entry -> entry.getKey().equalsIgnoreCase(request.getZosmfBaseUrl() + endpoint))
-            .findFirst().map(entry -> entry.getValue()).orElseThrow(IllegalStateException::new);
+                .findFirst().map(entry -> entry.getValue()).orElse(true);

```

Comment:
What is the reason for this change?

Reply:
it should not throw, this is an error. When the nedpoint is not in the map, it should try it, not throw

Reply:
I've added tests around this scenario. I missed this test the first time and then fixed it as part of sonar.